source/module_io/get_wf_lcao.cpp

```diff
@@ -28,157 +29,142 @@ void IState_Envelope::begin(const psi::Psi<double>* psid,
                             const K_Vectors& kv,
                             const double nelec,
                             const int nbands_istate,
-                            const std::vector<int>& out_band_kb,
+                            const std::vector<int>& out_wfc_norm,
+                            const std::vector<int>& out_wfc_re_im,
                             const int nbands,
                             const int nspin,
                             const int nlocal,
                             const std::string& global_out_dir)
 {
     ModuleBase::TITLE("IState_Envelope", "begin");
 
-    std::cout << " Perform |psi(band, r)| for selected bands." << std::endl;
+    std::cout << " Perform |psi(i, r)|, Re[psi(i, r)], Im[psi(i, r)] for selected bands (gamma only)." << std::endl;
 
-    int mode = 0;
-    if (nbands_istate > 0 && static_cast<int>(out_band_kb.size()) == 0)
-    {
-        mode = 1;
-    }
-    else if (static_cast<int>(out_band_kb.size()) > 0)
-    {
-        // If out_band_kb (bands_to_print) is not empty, set mode to 2
-        mode = 2;
-        std::cout << " Notice: INPUT parameter `nbands_istate` overwritten by `bands_to_print`!" << std::endl;
-    }
-
-    int fermi_band = 0;
-    int bands_below = 0;
-    int bands_above = 0;
-
-    this->bands_picked_.resize(nbands);
-    ModuleBase::GlobalFunc::ZEROS(bands_picked_.data(), nbands);
-
-    // (1)
-    // mohan update 2011-03-21
     // if ucell is odd, it's correct,
     // if ucell is even, it's also correct.
     // +1.0e-8 in case like (2.999999999+1)/2
+    const int fermi_band = static_cast<int>((nelec + 1) / 2 + 1.0e-8);
```

Comment:
I am concern with the use of magic number here, can this line be changed with function ceil() or something?